std/internal/phobosinit.d

```diff
@@ -0,0 +1,35 @@
+// Written in the D programming language.
+
+/++
+    The purpose of this module is to perform static construction away from the
+    normal modules to eliminate cyclic construction errors.
+
+    Copyright: Copyright 2011 - 2016
+    License:   $(HTTP www.boost.org/LICENSE_1_0.txt, Boost License 1.0).
+    Authors:   Jonathan M Davis, Kato Shoichi, Steven Schveighoffer
+    Source:    $(PHOBOSSRC std/internal/_phobosinit.d)
+  +/
+module std.internal.phobosinit;
+
+version(OSX)
+{
+    extern(C) void std_process_shared_static_this();
+
+    shared static this()
+    {
+        std_process_shared_static_this();
+    }
+}
+
+shared static this()
+{
+    import std.encoding;
```

Comment:
I have to say, the way this is done in `std.encoding` is horrific. -- You register the encoding class name *by string*, and then it uses the runtime to look up the class info via search of that string. THEN it instantiates it, and checks all the encodings that are supported (and then lets the GC collect it).

There MUST be a better way to do this...